identity-admin/webapp/components/search/StatusDropdown.test.tsx

```diff
@@ -0,0 +1,50 @@
+import { fireEvent, render } from '@testing-library/react';
+import StatusDropdown from './StatusDropdown';
+import { buildSearchUrl } from '../../utils/search-util';
+
+jest.mock('next/router', () => ({
+  useRouter: () => {
+    return {
+      query: {
+        status: 'active',
+        name: 'Bob',
+        email: 'bob@email.com',
+        sort: 'email',
+        sortDir: '1',
+      },
+    };
+  },
+}));
+
+describe('Status dropdown', () => {
+  const { location } = window;
+
+  beforeAll((): void => {
+    // @ts-ignore
+    delete window.location;
+    // @ts-ignore
+    window.location = {
+      href: '',
+    };
+  });
+
+  afterAll((): void => {
+    window.location = location;
+  });
+
+  it('redirects to search with status filter', async () => {
+    const { container } = render(<StatusDropdown />);
+    const select = container.firstChild;
+    if (select) {
+      expect(select).toHaveValue('active');
+      fireEvent.change(select, {
+        target: { value: 'locked' },
+      });
+      expect(window.location.href).toBe(
+        buildSearchUrl('1', 'locked', 'Bob', 'bob@email.com')
+      );
+    } else {
+      fail('Select element was not rendered');
```

Comment:
This one was new to me! I didn't know Jest had a `fail` method. But the Jest maintainers [don't want us using it](https://github.com/facebook/jest/issues/2129#issuecomment-454000295).

Why is there a risk that the `select` isn't rendered, though? Do we need this `if`/`else`?

Reply:
This is just because `querySelector` is typed like this:

```
querySelector<E extends Element = Element>(selectors: string): E | null;
```

The `else` case was just about appeasing the type system in the case that `querySelector` returned `null` (which should never have happened), so we can remove this with the change from using `container`